modules/reactor/src/meshgenerators/PeripheralRingMeshGenerator.C

```diff
@@ -150,46 +150,88 @@ PeripheralRingMeshGenerator::generate()
   if (hasMeshProperty<bool>("square_center_trimmability", _input_name))
     setMeshProperty("square_center_trimmability",
                     getMeshProperty<bool>("square_center_trimmability", _input_name));
-  // Calculate biasing terms
-  const auto main_peripheral_bias_terms =
-      biasTermsCalculator(_peripheral_radial_bias, _peripheral_layer_num);
-  const auto inner_peripheral_bias_terms =
-      biasTermsCalculator(_peripheral_inner_boundary_layer_params.bias,
-                          _peripheral_inner_boundary_layer_params.intervals);
-  // It is easier to create outer boundary layer inversely (inwards). Thus, 1.0 / bias is used here.
-  // However, the input parameter definition is not affected.
-  const auto outer_peripheral_bias_terms =
-      biasTermsCalculator(1.0 / _peripheral_outer_boundary_layer_params.bias,
-                          _peripheral_outer_boundary_layer_params.intervals);
 
-  const unsigned int total_peripheral_layer_num =
-      _peripheral_inner_boundary_layer_params.intervals + _peripheral_layer_num +
-      _peripheral_outer_boundary_layer_params.intervals;
   // Need ReplicatedMesh for stitching
   auto input_mesh = dynamic_cast<ReplicatedMesh *>(_input.get());
   if (!input_mesh)
     paramError("input", "Input is not a replicated mesh, which is required.");
   if (*(input_mesh->elem_dimensions().begin()) != 2 ||
       *(input_mesh->elem_dimensions().rbegin()) != 2)
     paramError("input", "Only 2D meshes are supported.");
+
+  _input_mesh_external_bid =
+      MooseMeshUtils::getBoundaryID(_input_mesh_external_boundary, *input_mesh);
+  // We check the element types of input mesh's external boundary here.
+  // We support both linear and quadratic sides (i.e., EDGE2 and EDGE3), but we cannot support mixed
+  // sides
+  auto side_list_tmp = input_mesh->get_boundary_info().build_side_list();
+  bool linear_side = false;
+  bool quadratic_side = false;
+  for (auto & side : side_list_tmp)
+  {
+    if (std::get<2>(side) == _input_mesh_external_bid)
+    {
+      const auto etype = input_mesh->elem_ptr(std::get<0>(side))->side_type(std::get<1>(side));
+      if (etype == EDGE2)
+        linear_side = true;
+      else if (etype == EDGE3)
+        quadratic_side = true;
+      else
+        paramError("input", "Input contains elements that are not supported by this generator.");
```

Comment:
Please add a `RunException` test for this.

Reply:
Does current MOOSE have an easy way to load/create a third-order 2D mesh to trigger this error?

Reply:
Tagging @roystgnr on this one since I'm not sure. He will likely know off the top of his head.

Reply:
I see that libmesh has a 1D `Edge4` element. I'll see if I can figure out what kind of element uses this.

Reply:
As far as I can tell from looking through libmesh source, `Edge4` is not used as a side in any element. I'll let Roy have the final word, though.

Reply:
Right, I do not see any complete third-order 2D elements there.

Reply:
We currently don't have any 3rd-order elements outside of 1D, so no way to add test coverage for this.

We might be adding some in the next year, so it's good that we're catching that possibility here, but not soon enough that I expect I'll remember this PR when we're finally able to add a test for it.